products/models.py

```diff
@@ -0,0 +1,67 @@
+from django.db import models
+
+class Product(models.Model):
+    name            =   models.CharField(max_length=45)
+    subcategory     =   models.ForeignKey('SubCategory', on_delete=models.CASCADE)
+    stock           =   models.IntegerField()
+    price           =   models.DecimalField(max_digits=10, decimal_places=2)
+    discount_rate   =   models.OneToOneField('DiscountRate', on_delete=models.CASCADE)
+    category        =   models.ForeignKey('Category', on_delete=models.CASCADE)
+    created_at      =   models.DateTimeField(auto_now_add=True)
+    updated_at      =   models.DateTimeField(auto_now=True)
+
+    class Meta:
+        db_table    = 'products'
+
+class Category(models.Model):
+    name            =   models.CharField(max_length=45)
+
+    class Meta:
+        db_table    = 'categories'
+
+class SubCategory(models.Model):
+    name            = models.CharField(max_length=45)
+    category        = models.ForeignKey('Category', on_delete = models.CASCADE)
+
+    class Meta:
+        db_table    = 'subcategories'
+
+class Allergy(models.Model):
+    name            = models.CharField(max_length=45)
+
+    class Meta:
+        db_table    = 'allergies'
+
+class ProductInformation(models.Model):
+    sale_unit       =  models.CharField(max_length=45)
+    weight_g        =  models.DecimalField(max_digits=3, decimal_places=2)
+    delivery_type   =  models.CharField(max_length=45)
+    packing_type    =  models.CharField(max_length=45)
+    instruction     =  models.TextField(null=True)
+    product         =  models.OneToOneField('Product' ,on_delete = models.CASCADE)
+    comment         =  models.CharField(max_length=100,null=True)
+    allergy         =  models.ManyToManyField('Allergy', through = 'AllergyProduct')
+
+    class Meta:
+        db_table = 'productinformations'
+
+class AllegyProduct(models.Model):
+    allergy             = models.ForeignKey('Allergy', on_delete=models.CASCADE)
+    productinformation  = models.ForeignKey('ProductInformation', on_delete=models.CASCADE)
+
+    class Meta:
+        db_table = 'allegyproducts'
+
+class Image(models.Model):
+    img_url         = models.CharField(max_length=500)
```

Comment:
```suggestion
    image_url         = models.URLField(max_length=2000)
```
필드명은 줄임말을 지양해주시고, URLField라는 것이 있습니다. 물론 컨벤션입니다. 필드효과는 장고의 form이라는 것을 써야 효과가 있는데 우리는 과정상 사용하지 않습니다. 그래서 기능상으로는 CharField와 같습니다.
그리고 url을 다룰때는 길이제한이 짧으면 좋지 않으므로, 최소 2000자 길이는 주도록 합시다.

Reply:
수정 완료 했습니다.